leapp/workflows/__init__.py

```diff
@@ -141,6 +149,12 @@ def serialize(cls):
             'phases': [phase.serialize() for phase in cls.phases],
         }
 
+
+    def is_valid_phase(self, phase=None):
+        if phase:
+            return phase in [phase_names(phase) for phase in self._phase_actors]
```

Comment:
you Need to flatten the list of tuples to be able to search in it

Reply:
Should be fine now.